test/db-migrations/mocha-setup.js

```diff
@@ -0,0 +1,52 @@
+const _log = level => (...args) => console.log(level, ...args);
+global.log = _log('[INFO]');
+
+global.assert = require('node:assert');
+const fs = require('node:fs');
+const slonik = require('slonik');
+const migrator = require('./migrator');
+
+async function mochaGlobalSetup() {
+  log('mochaGlobalSetup() :: ENTRY');
+
+  global.assert = assert;
+
+  global.sql = slonik.sql;
+
+  const { user, password, host, database } = jsonFile('./config/db-migration-test.json').default.database;
+  const dbUrl = `postgres://${user}:${password}@${host}/${database}`;
+  log('dbUrl:', dbUrl);
+  global.db = slonik.createPool(dbUrl);
+
+  const existingTables = await db.oneFirst(sql`SELECT COUNT(*) FROM information_schema.tables WHERE table_schema='public'`);
+  if(existingTables) {
+    console.log(`
+    Existing tables were found in the public database schema.  Reset the database before running migration tests.
```

Comment:
In what situations will this be needed? Is the database automatically reset when tests succeed? It'd be nice to have to manually reset only rarely.

Reply:
Resetting the database requires different permissions:

```sh
$ docker exec odk-postgres14 psql -U jubilant jubilant_test -c "
        DROP SCHEMA public CASCADE;
        CREATE SCHEMA public;
        GRANT ALL ON SCHEMA public TO postgres;
        GRANT ALL ON SCHEMA public TO public;
      "
ERROR:  must be owner of schema public
```

vs

```sh
$ docker exec odk-postgres14 psql -U postgres jubilant_test -c "
        DROP SCHEMA public CASCADE;
        CREATE SCHEMA public;
        GRANT ALL ON SCHEMA public TO postgres;
        GRANT ALL ON SCHEMA public TO public;
      "
NOTICE:  drop cascades to 47 other objects
...
GRANT
```

Reply:
Can we add the required permissions then? I don't think it'll work to have to manually reset after each test run.

Or can we do whatever we do in our current tests? We're able to reset the database there with our current permissions.

Reply:
> Can we add the required permissions then?

This would be an option, but it would mean either having different database permissions in tests vs production, or relaxing database permissions in production.

> I don't think it'll work to have to manually reset after each test run.

I'd recommend running before every test run, by default.  E.g.:

```sh
docker exec odk-postgres14 psql -U postgres jubilant_test -c "
        DROP SCHEMA public CASCADE;
        CREATE SCHEMA public;
        GRANT ALL ON SCHEMA public TO postgres;
        GRANT ALL ON SCHEMA public TO public;
      " && make test-db-migrations
```

Reply:
That feels a little tough to me, because it'll require looking up that code whenever we just want to run tests locally. Also, not everyone on the team is using the `odk-postgres14` Docker.

Is it possible to do `drop owned by current_user` instead? That's what we do in our current tests.

Reply:
If that'd work, I really think it'd make running these tests that much more convenient. If it's been working for our integration tests, hopefully it'd work in this case as well.

Reply:
> If it's been working for our integration tests, hopefully it'd work in this case as well.

I'm not sure this is a safe assumption - the migration tests are explicitly aiming to test at a different level to the integration tests.

That said, I think `drop owned by current_user` looks pragmatic on dev machines, and we can rely on CI to run migrations in a clean environment.